lib/test.js

```diff
@@ -58,8 +58,16 @@ function configure(sinon, config) {
             } catch (e) {
                 exception = e;
             }
-
-            if (typeof oldDone !== "function") {
```

Comment:
@ThomasChadzelek ,

Hmm... my thinking was that if you returned a promise that is pretty unambiguous... you could be accepting a function as an argument for some other reason (who knows why) -- as using promises you aren't required to follow the callback convention. 

But I don't attach much weight to that thought. Also, then you are right: I'm inconsistent, as the wrapped version of the callback has been passed in, and you can't tell if a promise will be passed back until after you make the call.

As I don't see any way to be consistent vs how I was thinking about it, I think you are right, but I'll leave the call to @fatso83 -- will change if he thinks its good.

 @fatso83?